public/scripts/game.js

```diff
@@ -78,23 +85,55 @@ class Game {
 		}, 3000);
 	}
 
-	hoverCheck(tile, mX, mY) {
+	tileHoverCheck(tile, mX, mY) {
 		if (this.getPlayerTurn() !== tile.player) {
 			return false;
 		}
 		const delta = Math.sqrt((mX - tile.x) ** 2 + (mY - tile.y) ** 2);
 		return delta < tile.d / 2;
 	}
 
-	// not dry
-	// reuse hover function somehow!
-	clickTile(e) {
+	discHoverCheck(tile, mX, mY) {
+		const delta = Math.sqrt((mX - tile.x) ** 2 + (mY - tile.y) ** 2);
+		return delta < tile.d / 2;
+	}
+
+	resetClick() {
 		for (const tile of this.tokens) {
 			tile.clicked = false;
 		}
 		for (const disc of this.discs) {
 			disc.previewMode = false;
 		}
+	}
+
+	checkLegalMove(token, disc) {
+		return token.tokenInfo.moveableTo.some((legalMoves) => legalMoves.id === disc.discInfo.id);
+	}
+
+
+	clickTile(e) {
+		const clickedToken = this.tokens.find((t) => t.clicked);
+		const nextDiscPosition = this.discs.find((d) => d.hovering);
+
+		// checks if player is trying to make a move and validates it (and makes move if legal)
+		if (clickedToken && nextDiscPosition && this.checkLegalMove(clickedToken, nextDiscPosition)) {
+			makeMove(this.tokens, this.discs, this.gameHistory, clickedToken, nextDiscPosition)
+				.then((result) => {
+					// console.log(result, 'result')
+					// createNewGameWithNewResult()
+				})
+				.catch((error) => {
+					console.log(error);
+					this.errorMsg = error;
+
+					// draws error message in nice color
+					setTimeout(() => {
+						this.errorMsg = null;
+					}, 5000);
+				});
```

Comment:
I bet you can extract this to be a more generic error handler, that we can re-use for all the other requests (and to clean up the code a bit)